File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<pair<int,int>>pairSet;
        for(int i = 0; i < nums.size(); i++){
            pairSet.push_back({nums[i], i});
        }
        
        sort(pairSet.begin(), pairSet.end());
 
        int i = 0, j = pairSet.size() - 1;
        while(i < j){
            long long sum = pairSet[i].first + pairSet[j].first;
            if(sum == target){
                return {pairSet[i].second, pairSet[j].second};
            }else if(sum < target){
                ++i;
            }else{
                --j;
            }
        }
        return {-1, -1};
    }
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        int []res = new int[2];
        
        Map<Integer, Integer> mp = new HashMap<>();
        
        int n = nums.length;
        
        for(int i = 0; i < n; i++){
            
            int k = target - nums[i];
            
            if(mp.containsKey(k)){
                
                res[0] = mp.get(k);
                
                res[1] = i;
                
                break;
                
            }
            
            mp.put(nums[i], i);
            
        }
        
        return res;
        
    }
}